### PR TITLE
Add editorconfig file

### PR DESCRIPTION
**Pull request by fsateler**
The first commit is the interesting one. The second one just applies the results willi-nilly to see how much difference do we get. It should be removed before merging.

Some cases cannot be expressed in the (simple) editorconfig language: table-like forms are lost. The extra indentation added to inner methods is not possible to express either AFAICS: to the editorconfig service, it is a new statement and thus should be indented at the method level.

The rest of the differences look very reasonable to me.

### CODE REVIEW

File: MoreLinq.sln [hunk 1]
```diff
@@ -1,9 +1,10 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26228.4
+VisualStudioVersion = 15.0.26403.7
```
Comment (atifaziz) [inline MoreLinq.sln]: This change is unrelated.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.xml]
+indent_style = space
+
+[*.cs]
+indent_style = space
+indent_size = 4
+csharp_new_line_before_open_brace = properties,control_blocks,types,methods,anonymous_types,object_collection_array_initalizers
```
Comment (atifaziz) [inline .editorconfig]: I would drop everything `charp_*` as this is subjective and should be discussed and submitted as a separate PR.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,21 @@
+root = true
+
+[*]
+end_of_line = crlf
```
Comment (atifaziz) [inline .editorconfig]: This shouldn't be necessary with new-line normalization in `.gitattributes`. Also new-line terminators vary with platforms.

**Comment by fsateler** [comment]
> Please revert e575e00 as it has destroyed a lot hand-formatting of code for readability. The problem with global application of styles is exactly this

Agreed. Not all styling can be applied automatically. The point of that commit is merely to show what sort of formatting would be automatically applied. As I said in the PR message, that commit should not be merged.

> I would drop everything `charp_*` as this is subjective and should be discussed and submitted as a separate PR.

Well, my primary goal was to reduce the number of style corrections (such as [this one](https://github.com/morelinq/MoreLINQ/pull/331#discussion_r125746564)). As such, I tried to preserve the current conventions as much as it is possible to express in the relatively simple set of editorconfig knobs. In other words, this is a simple first step towards fixing #247. If I missed a knob, or set one incorrectly, I'd like to see that fixed too.

File: MoreLinq.sln (hunk 1 above)
Comment (fsateler) [reply MoreLinq.sln]: This was automatically inserted. Unfortunately VS encodes the version in the sln file, so at some point this number should be bumped? Why not now?

**Comment by atifaziz** [comment]
> Well, my primary goal was to reduce the number of style corrections (such as this one). As such, I tried to preserve the current conventions as much as it is possible to express in the relatively simple set of editorconfig knobs. In other words, this is a simple first step towards fixing #247. If I missed a knob, or set one incorrectly, I'd like to see that fixed too.

Agreed for going forward provided we revert e575e00a277aadf6da97f687601b01a3b105535a on this PR so change net out on a squash-merge

File: MoreLinq.sln (hunk 1 above)
Comment (atifaziz) [reply MoreLinq.sln]: I'm not going to pick on it but changes in a diff should be related to the commit message and anything else is a distraction. Something like this can be avoided with a `git add -p`.

**Comment by atifaziz** [comment]
The PR branch now has a conflict with master that will disappear if e575e00a277aadf6da97f687601b01a3b105535a is reverted.

**Comment by fsateler** [comment]
I have reverted the `csharp_*` directives, and added a few more changes on top (each with explanatory commit). PTAL.

**Comment by fsateler** [comment]
> I don't think 0296bfa had the intended effect. It reverted the opposite change.

Doh! 🤦‍♂️  I got confused with the revert and `reset -p`
